server/README.md

```diff
@@ -8,6 +8,36 @@ The server will expose a package repository over http. It is located at `/[arch]
 
 The repository follows the format of an ordinary Arch Linux repository, just the way pacman expects it. The name of the actual repository (determined by the .db file) can be [configured](#configuration) on the container, and is `serene` by default. The package archive format is `.tar.zst`.
 
+### Package Signing
+For security reasons the packages built by the build server can be signed. Whilst the signature is no guarantee the built package does not contain any malware it still verifies the package was built on your build server.
+This is useful when you host your repository without a ssl certificate (which is generally discouraged) or when there are mirrors which redistribute packages from your build server. 
+
+To enable package signing you'll have to mount a gpg private key into the server container. The gpg private key can be exported like this:
+```shell
+# export private key into `private-key.sec.asc` in armored ascii format
+gpg --armored --export-secret-key <key-id> --output private-key.sec.asc
+```
+
+Now, when [deploying](#deployment) you'll have to add an additional volume bind to the `docker-compose.yml`: 
+```yml
+# docker-compose.yml
+
+services:
+  serene:
+    # ... remaining service definition
+    volumes:
+      - /path/to/private-key.sec.asc:/app/sign-private-key.sec.asc
+      # ... other volumes
+      
+```
+
+Should your private key be protected by a password you'll have to additionally [configure the password](#configuration).
+
+To enable package verification using pacman you'll need to download the public key from the `/key` api endpoint and [import it into your pacman keys](https://wiki.archlinux.org/title/Pacman/Package_signing#Adding_unofficial_keys).
```

Comment:
also add a note that only newly built packages will be signed, and old packages will not be

also state that they should remove the optional, trustall tags from the repo def